airbyte-cdk/python/airbyte_cdk/sources/declarative/decoders/json_decoder.py

```diff
@@ -17,8 +18,43 @@ class JsonDecoder(Decoder):
 
     parameters: InitVar[Mapping[str, Any]]
 
-    def decode(self, response: requests.Response) -> Union[Mapping[str, Any], List[Any], Any]:
+    def decode(self, response: requests.Response) -> Generator[Mapping[str, Any], None, None]:
         try:
-            return response.json()
+            body_json = response.json()
+            if not isinstance(body_json, list):
+                body_json = [body_json]
+            yield from body_json
         except requests.exceptions.JSONDecodeError:
-            return {}
+            yield {}
+
+
+@dataclass
+class IterableDecoder(Decoder):
+    """
+    Decoder strategy that returns the string content of the response, if any.
+    """
+
+    is_stream_response = True
+    parameters: InitVar[Mapping[str, Any]]
+
+    def decode(self, response: requests.Response) -> Generator[Mapping[str, Any], None, None]:
+        # TODO: how to handle simple string in extractor;
+        #  see list_users in iterable:: response.body == b'user1@example.com\nuser2@example.com'
+        #  possible option: we can wrap strings directly into records => {"record": {line.decode()}}
+        for line in response.iter_lines():
+            yield {"record": line.decode()}
+
+
+@dataclass
+class JsonlDecoder(Decoder):
+    """
+    Decoder strategy that returns the json-encoded content of the response, if any.
+    """
+
+    is_stream_response = True
+    parameters: InitVar[Mapping[str, Any]]
+
+    def decode(self, response: requests.Response) -> Generator[Mapping[str, Any], None, None]:
+        # TODO???: set delimiter? usually it is `\n` but maybe it would be useful to set optional?
```

Comment:
good idea, but ok to leave as a follow up as it's not currently needed. can you create a follow up issue?

Reply:
created https://github.com/airbytehq/airbyte-internal-issues/issues/8436